src/debug.c

```diff
@@ -1819,23 +1821,115 @@ void closeDirectLogFiledes(int fd) {
 }
 
 #ifdef HAVE_BACKTRACE
+#define BACKTRACE_MAX_SIZE 100
+#ifdef __linux__
+#include <sys/prctl.h>
+#include <sys/syscall.h>
+#include <dirent.h>
+
+static pid_t *get_ready_to_signal_threads_tids(pid_t pid, int sig_num, size_t *tids_len_output);
+
+#define MAX_BUFF_LENGTH 256
+typedef struct {
+    char thread_name[16];
+    int trace_size;
+    pid_t tid;
+    void *trace[BACKTRACE_MAX_SIZE];
+} stacktrace_data;
+
+static void *collect_stacktrace_data(void) {
+    /* allocate stacktrace_data struct */
+    stacktrace_data *trace_data = zmalloc(sizeof(stacktrace_data));
```

Comment:
```suggestion
    stacktrace_data *trace_data = zcalloc(sizeof(stacktrace_data));
```
if `prctl(PR_GET_NAME, trace_data->thread_name);` fail, `trace_data->thread_name` will be uninitialized.

Reply:
possible failures of `prctl` in this case are:

> EFAULT
arg2 is an invalid address.
EINVAL
The value of option is not recognized.
EINVAL
arg2 is not valid value for this option.

so if it fails, an uninitialized string is our last concern.

Reply:
@meiravgri How can we make sure it's safe to allocate memory in a signal handler on all platforms? See the `signal-safety(7)` man page. I think this is also true specifically for jemalloc.